### ISSUE TITLE
MVP discussion

### ISSUE THREAD

**Issue by zaviermiller** [hack4impact-utk/compassion-ministries#127]
### Discussed in https://github.com/hack4impact-utk/compassion-ministries/discussions/120
I decided to put this here so we can track progress

<div type='discussions-op-text'>

<sup>Originally posted by **zaviermiller** February 21, 2024</sup>
# Background
As the deadline for our MVP quickly approaches, we need to understand and agree what workflows and features are apart of the MVP. The way @zaviermiller and @andlrutt have thought about it is what needs to be done in order for the app to be usable for checking in volunteers at events, as this is the meat of the application.

# Overview
Before getting into exactly what needs to be done for our MVP, let's take a high-level look at the workflows we need.

## Event management
In order to check volunteers into events, users need to be able to create, update, view, and delete events. When listing all events, we should create a simple date range input that allows users to specify events that take place between this range. When viewing a single event, all the information should be displayed from the event, as well as all volunteers that were present at the event.

For now, recurring events are probably not necessary and can be delayed; making events recurring is just a nice-to-have the introduces a lot of complexity into these workflows, and while we should build the MVP event system with recurring events in mind, we don't need to build recurring events yet.

## Volunteer management
Before a volunteer can be checked in, they'll need to actually exist in the system. As a part of the MVP users will need to be able to create, update, view, and delete volunteers. When viewing a single volunteer, all the information should be displayed from the volunteer, as well as the events that the volunteer attended. Also, the single volunteer page should display a list of the roles a volunteer is verified for.

One thing to note here is that there are actually two places volunteers can be created. In a dedicated create volunteer page and during the check in flow. I'll talk about that more in the check in section

## Organization management
This is simple, users need to be able to create, update, view, and delete organizations.

## Role verification
When checking volunteers in to more sensitive roles, users need to be able to vouch for a volunteer to fulfill that role. Users also need to be able to update a verification, and remove a verification from a user

Similarly to the volunteer system above, this role verification can happen both during check in and outside of an event, so the system will need to support both.

## Background checks
Similarly to role verifications, volunteers in sensitive positions are subject to background checks. We would like the ability to kick these checks off from the system, and be able to show the status of background checks for a given volunteer.

This is likely in the MVP, but there are decisions that need to be made about what background check provider to use before we can implement this system. For now, we'll have to move forward without it, but we should implement it ASAP.

## Check in
Finally, users should be able to check in volunteers at specific events. This flow is quite complex, and is outlined in the diagram below.

![image](https://github.com/hack4impact-utk/compassion-ministries/assets/17971951/55b986f9-7b4f-4741-92f6-d32efeaec01e)

We will need to implement this system fully to the best of our ability. The diagram above is lacking one part of the flow that should be mentioned here: verifying a volunteer for a role. After the verified for role check, if the volunteer is not verified, we should ask the user if they wish to verify the user for that role.

## Summary
That's the high-level overview of the features needed for the MVP. Please feel free to add to the discussion if you believe anything is missing.

>[!NOTE]
> While @zaviermiller and @andlrutt technically are running the project and will inherently have the most knowledge of the product, every team member should be intimately familiar with the product we are building. The more people that are familiar with what we're doing, the less likely features are missed.

# In-depth checklist
This section will contain a checklist of tasks and their corresponding issues. We won't worry about keeping the checklist maintained, but it should give us a good idea of how far along we are at the end of February. Each checklist is sorted in order of relative importance. Also, some items won't have issues. That's ok, we will add those.

## Event management
- [ ] form for creating and editing events - #95 
- [x] view for single events - #96
- [x] event detail - #94 
- [ ] views for creating and updating an event -
- [ ] hook up views to backend -
- [x] backend support for creating an event - #77 
- [ ] backend support for getting a single event - #117 
- [ ] backend support for getting volunteers at given event - #116 
- [ ] backend support for listing events between dates - #79 
- [ ] view for listing events between dates - 
- [x] event list - #71
- [ ] frontend support for deleting events -
- [ ] backend support for updating an event - 
- [ ] backend support for deleting an event - 

## Volunteer management
- [ ] form for creating and editing volunteers - #73 
- [ ] view for single volunteer - 
- [x] volunteer detail - #88
- [ ] views for creating and updating a volunteer - #111 
- [ ] hook up views to backend - #90 
- [x] backend support for creating a volunteer - #9 
- [x] backend support for getting a single volunteer - #12
- [x] backend support for getting events a volunteer attended - #26 
- [x] backend support for getting all volunteers - #11 
- [ ] view for listing all volunteers (?) -
- [x] volunteer list - #69
- [ ] frontend support for deleting volunteers -
- [x] backend support for updating a volunteer - #10 
- [x] backend support for deleting a volunteer - #16

## Organization management
- [ ] form for creating and editing organization - #93 
- [ ] views for creating and updating an ogranization - #112 
- [x] org detail component - #99
- [ ] single org view - 
- [ ] hook up views to backend - #98 #100 
- [x] backend support for creating an organization - #13
- [x] backend support for getting all organizations - #15
- [ ] view for listing all organizations -
- [x] organization list - #92 
- [ ] frontend support for deleting organizations -
- [x] backend support for updating an organization - #14 
- [x] backend support for deleting an organization - #16

## Role verification
- [x] backend support for creating/updating role verifications - #27
- [x] backend support for removing role verifications - #28 
- [x] role verification form - #67 
- [ ] hook up role verification form to backend - #89 
- [ ] put role verification form on single vol view - #89 

## Background check
- [ ] make decision on service
- [ ] implement 1
- [ ] implement 2
- [ ] implement 3
- [ ] implement 4
- [ ] implement 5
- [ ] implement 6
- [ ] implement 7

## Check in
- [x] backend support to checkin a volunteer - #23
- [x] backend support to un-checkin a volunteer - #25 
- [x] check in form - #72 
- [x] check in view - #110 
- [x] hook up to backend - #97 
- [ ] autocomplete role, org, and other info from volunteer
- [ ] create volunteer if it doesnt exist already
- [ ] put role verification confirmation on submit when volunteer is not verified for role

# Summary
There is no way I remembered everything. There is a lot left to do, so we definitely need to pick up the pace a bit</div>
